### ISSUE TITLE
"Searching..." overlay does not automatically disappear once the search is complete

### ISSUE THREAD

**Issue by kevinaboos** [moxin-org/moxin#208]
After searching for a term on the "Discover" screen, the `Searching...` animation will be shown forever until the user scrolls on the screen. 

See screenshot below, which will remain on the screen indefinitely until the user scrolls up or down.

I assume the problem is related to something like a forgotten call to `redraw()` after the search results have been fully returned.

![image](https://github.com/user-attachments/assets/3f8ceaed-0987-43d7-9355-3ab3537176ed)

**Comment by CouldBeFree** [comment]
I want to resolve this issue, but I face some problems running this project.
I have got this ```error: failed to run custom build command for `wasmedge-sys v0.19.2```
I have tried to add this library to the cargo.toml file, but with no success
```
[dependencies]
moxin-protocol = { path = "moxin-protocol" }
moxin-backend = { path = "moxin-backend" }
moxin-fake-backend = { path = "moxin-fake-backend" }
wasmedge-sys = {version = "0.19.2"} -> New line
```

**Comment by kevinaboos** [comment]
Hi @CouldBeFree, thanks for looking into this! The error you're getting means that you haven't installed WasmEdge, or the build system cannot find your installation of WasmEdge. 

Have you followed the build & setup instructions in the readme? https://github.com/moxin-org/moxin/tree/dev?tab=readme-ov-file#building-and-running
If you have and it's not working for you, please let us know:
* which OS you're on
* dump out a `tree` view of the WasmEdge directory
* dump out the values of all environment variables 

Also, make sure to use the latest `dev` branch, `main` is a bit old.

**Comment by CouldBeFree** [comment]
@kevinaboos thank you for your support. I can try to resolve this issue

**Comment by Guocork** [comment]
I am using Ubuntu, and initially, I also encountered this problem. When I was fixing it, I adjusted the order of the [logic code](https://github.com/moxin-org/moxin/blob/main/src/landing/model_list.rs#L156-L162) which resolved the issue, but the logic became incorrect. I added a `redraw()` function, which solved the problem, but the searching disappeared very slowly. 
[https://github.com/moxin-org/moxin/pull/211](https://github.com/moxin-org/moxin/pull/211).

**Comment by joulei** [comment]
We'll review this behavior in Makepad:
https://github.com/moxin-org/moxin/pull/211#issuecomment-2290910793

**Comment by kevinaboos** [comment]
> @kevinaboos thank you for your support. I can't reproduce this bug, searching disappeared as expected
> 
>  Screen.Recording.compressed.mp4

yep, it appears to be working on macOS but not on other platforms, indicating a bug in the UI framework itself. Thanks for helping us test.
